api/v1beta1/keystoneapi_types.go

```diff
@@ -183,6 +185,70 @@ type PasswordSelector struct {
 	// +kubebuilder:default="AdminPassword"
 	// Admin - Selector to get the keystone Admin password from the Secret
 	Admin string `json:"admin"`
+
+	// OIDCClientSecret - Selector to get the IdP client secret from the Secret
+	KeystoneOIDCClientSecret string `json:"keystoneOIDCClientSecret"`
+
+	// OIDCCryptoPassphrase - Selector to get the OIDC crypto passphrase from the Secret
+	KeystoneOIDCCryptoPassphrase string `json:"keystoneOIDCCryptoPassphrase"`
+}
+
+type KeystoneFederationSpec struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// Enablement of Federation configuration
+	EnableFederation bool `json:"enableFederation,omitempty"`
```

Comment:
Not needed then - just check for len(instance.spec.OIDCFederation) >0

Reply:
Right now I'm going to leave the toggle for testing. It also follows the pattern of Secure RBAC in that the deployer can explicitly enable or disable it.

Reply:
OK - just realize of course that the reason upstream tests are failing is because you are not checking for len(instance.spec.OIDCFederation)>0